clusterctl/cmd/create.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/cluster-api/errors"
+)
+
+type CreateOptions struct {
+	Cluster string
+	Machine string
+	Extras  string
+}
+
+var co = &CreateOptions{}
+
+var createCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create kubernetes cluster",
+	Long:  `Create a kubernetes cluster with one command`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if co.Cluster == "" {
+			glog.Error("Please provide yaml file for cluster definition.")
+			cmd.Help()
+			os.Exit(1)
```

Comment:
We are doing the following:
```
cmd.Help()
os.Exit(...)
```

in 4 locations in the Create command and there are similar sections in Delete and Register commands. Theoretically this number will grow as the scope of this application increases. Is there a simplification we can make so that it is only called in a single place for the various error scenarios?

Reply:
Good point. DOne